ISSUE-27: Fix docker mongo setup

carlan opened the pull request:
This is the fix for the mongodb initialization. I haven't tested on MacOS and Windows, it's working properly on *nix based OS's.

What was done:
- Updated server docker-compose.yml to add container name and new volume
- Updated main README.md to add a note on MONGO_HOST configuration
- Added mongo-init.sh to initialize mongodb container

Fixes #27 💚

## Type of change

- [x] Bug fix

## PR Checklist:

- [x] Merged the current development branch before submit a PR
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

carlan committed:
ISSUE-27: Fix docker mongo setup

- Updated server docker-compose.yml to add container name and new volume
- Updated main README.md to add a note on MONGO_HOST configuration
- Added mongo-init.sh to initialize mongodb container

Fixes #27 💚

carlan committed:
Merge branch 'master' into bugfix/issue-27

w3cj commented on server/mongo-init.sh:
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+echo '========================================================================='
```
This is not required and happens by default... I removed all containers / images and db-data and was able to start up everything without this script.

The main fix that is needed is this:
* An init script on the API container that waits to see the message `waiting for connections on port 27017` before starting itself.

brilvio replied:
For the waiting connections I made a PR #39

carlan replied:
> The main fix that is needed is this:
> 
> * An init script on the API container that waits to see the message `waiting for connections on port 27017` before starting itself.

See my last comment below.

w3cj reviewed (changes requested):
See comment.

carlan committed:
ISSUE-27: Fix docker mongo setup

- Rolled back previous changes after feedback
- Updated docker-compose.yml
- Updated gitignore
- Added initialization scripts for db and api
- Added placeholder file

Fixes #27 💚

carlan committed:
Merge branch 'master' into bugfix/issue-27

carlan commented:
:wave: I had some time to work on this today. 

I updated this PR to undo the changes I've made previously and introduce some new files to solve the issue. It's working fine on *nix. I'm curious to see it running on Mac.

~One downside, the placeholder file docker-data/mongod.log will need to be present before hand otherwise docker will create it as a directory and we need it as a file. I had to update the gitignore and check in the docker-data and the placeholder file.~ EDIT: this isn't relevant with the latest changes.

Let me know your thoughts on it.

EDIT: I saw you had a bit of hard time understanding/remembering (no offence intended) what's going on. I should have added a comment explaining what I've done. Sorry about that.
The solution I've applied for this is rudimentar simple, I believe as you wanted/mentioned on previous stream. On db container I'm exporting the mongo logs in a volume, without it the server api wouldn't have a way to check if the database is ready or not based on the string. On server api container, I'm reading the log from the same volume and waiting for the string before continue.

carlan committed:
ISSUE-27: Fix docker mongo setup

- Rolled back previous changes on gitignore
- Updated docker-compose.yml
- Updated db and api initialization scripts
- Deleted placeholder file

Fixes #27 💚
